netstate/auth/process_api_key.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package auth
+
+import  (
+	"os"
+	"net/http"
+	"crypto/subtle"
+	
+	"github.com/spf13/viper"
+)
+
+func InitializeHeaders() *http.Header {
+	// mock HTTP request headers
+
+	httpHeaders := http.Header {
+		"Accept-Encoding": {"gzip, deflate"},
+		"Accept-Language": {"en-US,en;q=0.9"},
+		"X-Api-Key":         {"12345"},
+		"Cache-Control":    {"max-age=0"},
+		"Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8"} ,
+		"Connection":      {"keep-alive"},
+	}
+	return &httpHeaders
+}
+
+func setEnv() {
+	// if flag is not set, we'll set the env
+	// desgined to match xApiKey
+
+	viper.SetEnvPrefix("API") 
+	os.Setenv("API_KEY", "12345")
+	viper.AutomaticEnv()
+}
+
+func ValidateApiKey(header string)(bool) {
```

Comment:
looks like this is worth running through `gofmt`. you can do `gofmt -w process_api_key.go`

Reply:
didn't know you can do that!  got it.